### PR TITLE
Adding Bricklink Add to Cart

### PR DESCRIPTION
**Pull request by robarchibald**

### CODE REVIEW

File: bricklinkuser/api.go [hunk 1]
```diff
@@ -35,12 +36,28 @@ func NewClient(cred *credentials.BrickLinkUser) (*Client, error) {
...
+	if resp == nil || resp.Body == nil || err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	l := &LoginReturn{}
+	if err := json.NewDecoder(resp.Body).Decode(l); err != nil {
+		return err
+	}
+	if l.ReturnCode != 0 && l.ReturnMessage != "" {
```
Comment (andrewarchi) [inline bricklinkuser/api.go]: BrickLink JS does not check the return message, so please change to only `l.ReturnCode != 0`. BL only uses it to display info when the return code is non-zero.

File: brickset/brickset.go [hunk 2]
```diff
@@ -33,6 +33,10 @@ func (c *Client) makeRequest(method string, body string, result interface{}) err
 	if err != nil {
 		return err
 	}
+	// for debugging. Remove
+	//var buf bytes.Buffer
+	//buf.ReadFrom(resp.Body)
+	//fmt.Println(buf.String())
```
Comment (andrewarchi) [inline brickset/brickset.go]: Remove?

File: bricklinkuser/store.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
+package bricklinkuser
+
+import (
+	"encoding/json"
+	"io"
+	"net/url"
+)
+
+// CartItemSimple is a simplified representation of an item used to add the item to a cart
+type CartItemSimple struct {
```
Comment (andrewarchi) [inline bricklinkuser/store.go]: How did you arrive at this simplified struct? What is `sourceType`?

**Review by thaliaarchi — CHANGES_REQUESTED** [review]
Looks good. I just have a couple of comments before merging. Thanks!

File: bricklinkuser/store.go (hunk 3 above)
Comment (robarchibald) [reply bricklinkuser/store.go]: This is what the web UI passes in when you add an item to the cart. Sorry, I don’t know what it means, just found what it was passing in.

**Pull request merged**